policy/attestation_policy.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2021 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policy
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/checks/evaluation"
+	sce "github.com/ossf/scorecard/v4/errors"
+	"github.com/ossf/scorecard/v4/remediation"
+)
+
+//nolint:govet
+type AttestationPolicy struct {
+	// PreventBinaryArtifacts : set to true to require that this project's SCM repo is
+	// free of binary artifacts
+	PreventBinaryArtifacts bool `yaml:"preventBinaryArtifacts"`
+
+	// AllowedBinaryArtifacts : List of binary artifact paths to ignore
+	// when checking for binary artifacts in a repo
+	AllowedBinaryArtifacts []string `yaml:"allowedBinaryArtifacts"`
+
+	// EnsureNoVulnerabilities : set to true to require that this project is free
+	// of vulnerabilities, as discovered from the OSV service
+	EnsureNoVulnerabilities bool `yaml:"ensureNoVulnerabilities"`
+
+	// EnsurePinnedDependencies : set to true to require that this project pin dependencies
+	// by hash/commit SHA
+	EnsurePinnedDependencies bool `yaml:"ensurePinnedDependencies"`
+
+	// EnsureCodeReviewed : set to true to require that the most recent commits in
+	// this project have gone through a code review process
+	EnsureCodeReviewed bool `yaml:"ensureCodeReviewed"`
```

Comment:
Shouldn't these be more granular? For example, how can a user ignore pinned dependencies for a specific file or for a specific library? For Code-Review, lets also expose options to set `minApprovals` and `requiredApprovers`?